platform/android/export/export.cpp

```diff
@@ -1896,6 +1896,55 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		return run_icon;
 	}
 
+	static String get_adb_path() {
+		String exe_ext = "";
+		if (OS::get_singleton()->get_name() == "Windows") {
+			exe_ext = ".exe";
+		}
+		String sdk_path = EditorSettings::get_singleton()->get("export/android/android_sdk_path");
+		return sdk_path.plus_file("platform-tools/adb" + exe_ext);
+	}
+
+	static String get_apksigner_path() {
+		String exe_ext = "";
+		if (OS::get_singleton()->get_name() == "Windows") {
+			exe_ext = ".bat";
+		}
+		String apksigner_command_name = "apksigner" + exe_ext;
+		String sdk_path = EditorSettings::get_singleton()->get("export/android/android_sdk_path");
+		String apksigner_path = "";
+
+		Error errn;
+		String build_tools_dir = sdk_path.plus_file("build-tools");
+		DirAccessRef da = DirAccess::open(build_tools_dir, &errn);
+		if (errn != OK) {
+			print_error("Unable to open Android 'build-tools' directory.");
+			return apksigner_path;
+		}
+
+		// There are additional versions directories we need to go through.
+		da->list_dir_begin();
+		String sub_dir = da->get_next();
+		while (!sub_dir.empty()) {
+			if (!sub_dir.begins_with(".") && da->current_is_dir()) {
+				// Check if the tool is here.
+				String tool_path = build_tools_dir.plus_file(sub_dir).plus_file(apksigner_command_name);
+				if (FileAccess::exists(tool_path)) {
+					apksigner_path = tool_path;
+					break;
+				}
+			}
+			sub_dir = da->get_next();
+		}
+		da->list_dir_end();
```

Comment:
How confident are we that the SDK structure will stay the same? Parts of it have changed every now and then causing breakage in build scripts, and it would be annoying to have users unable to export APKs because apksigner moved someplace else and our hardcoded detection logic can't find it.

Reply:
That directory structure has been stable for the past 4 years ([since revision 24.0.3](https://developer.android.com/studio/releases/build-tools)), so I don't expect it to change soon.

Nonetheless, we can add a setting entry specific to the `apksigner` command, and only use the hardcoded logic above as a fallback in case the setting is not specified.
As we move more toward `AAB` and custom build, that setting will be less and less relevant (since we leverage the gradle build process instead for these).

Reply:
I guess let's go without extra setting for now and see how it evolves.